ctl/src/main.rs

```diff
@@ -27,11 +28,14 @@ fn main() {
   let config_file = matches.config;
 
   let config = Config::load_from_path(config_file.as_str()).expect("could not parse configuration file");
-  let mut channel: Channel<ConfigMessage,ConfigMessageAnswer> =
+  let connector = || -> Result<Channel<ConfigMessage,ConfigMessageAnswer>,io::Error> {
```

Comment:
I'm not a fan of this closure here. I'd rather have `upgrade` explicitely create the channel

Reply:
Would a function `fn connect(config) -> Result<Channel<..>, io::Error>` be better? which means config instance would have to be passed into the upgrade command as such  `upgrade(connection, &config)`?